src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -570,16 +570,11 @@ private String key() {
 
     /** Handles normal build completion (including errors) but also handles the case that the flow did not even start correctly, for example due to an error in {@link FlowExecution#start}. */
     private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
-        boolean nullListener = false;
-        synchronized (getLogCopyGuard()) {
-            nullListener = listener == null;
+        try {
             setResult(r);
```

Comment:
Doesn't `setResult(r);` still need to be guarded?

Reply:
I see no reason why it would.

Reply:
Other parts of this file have result setting inside the the synch block.

Reply:
Only incidentally I think.